tdrs-backend/tdpservice/parsers/models.py

```diff
@@ -67,3 +67,71 @@ def __str__(self):
     def _get_error_message(self):
         """Return the error message."""
         return self.error_message
+
+class DataFileSummary(models.Model):
+    """Aggregates information about a parsed file."""
+
+    class Status(models.TextChoices):
+        """Enum for status of parsed file."""
+
+        PENDING = "Pending"  # file has been uploaded, but not validated
+        ACCEPTED = "Accepted"
+        ACCEPTED_WITH_ERRORS = "Accepted with Errors"
+        REJECTED = "Rejected"
+
+    status = models.CharField(
+        max_length=50,
+        choices=Status.choices,
+        default=Status.PENDING,
+    )
+
+    datafile = models.ForeignKey(DataFile, on_delete=models.CASCADE)
+
+    #  eventually needs a breakdown of cases (accepted, rejected, total) per month
+    #  elif qtr2 jan-mar
+    #  elif qtr3 apr-jun
+    #  elif qtr4 jul-sept
+
+    case_aggregates = models.JSONField(null=True, blank=False)
+    """
+    # Do these queries only once, save result during creation of this model
+    # or do we grab this data during parsing and bubble it up during create call?
+    {
+        "Jan": {
+            "accepted": 100,
+            "rejected": 10,
+            "total": 110
+        },
+        "Feb": {
+            "accepted": 100,
+            "rejected": 10,
+            "total": 110
+        },
+        "Mar": {
+            "accepted": 100,
+            "rejected": 10,
+            "total": 110
+        }
+    """
+
+    def set_status(self, errors):
+        """Set and return the status field based on errors and models associated with datafile."""
+        # to set rejected, we would need to have raised an exception during (pre)parsing
+        # else if there are errors, we can set to accepted with errors
+        # else we can set to accepted (default)
+        if errors == {}:  # This feels better than running len() on `errors`...but is it a dict vs list?
```

Comment:
Assumption here that we have a dict and not just the list of parserError objects. Could be convinced if we feel it might be worth refactoring other areas to be more functional but this feels good enough.